cmd/syncthing/gui.go

```diff
@@ -1023,6 +1026,81 @@ func (s *apiService) getSystemLogTxt(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+type fileByte struct {
+	name string
+	buf  bytes.Buffer
+}
+
+func (s *apiService) getSupportBundle(w http.ResponseWriter, r *http.Request) {
+	var arrayFile []fileByte
+
+	jsonConfig, _ := json.MarshalIndent(getRedactedConfig(s), "", "  ")
+	arrayFile = addBufferFile(arrayFile, "config.json", jsonConfig)
+
+	jsonLog, _ := json.MarshalIndent(s.systemLog.Since(time.Time{}), "", "  ")
+	arrayFile = addBufferFile(arrayFile, "log.json", jsonLog)
+
+	jsonError, _ := json.MarshalIndent(s.guiErrors.Since(time.Time{}), "", "  ")
+	if string(jsonError) != "null" {
```

Comment:
I'd suggest

```
if errs := s.guiErrors.Since(time.Time{}); len(errs) > 0 {
  // do marshalling etc
}
```

instead of comparing to the "null" string here.